fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/EraseResourceDAO.java

```diff
@@ -0,0 +1,351 @@
+            call.setString(2, eraseDto.getLogicalId());
+            call.registerOutParameter(3, Types.BIGINT);
+            call.execute();
+
+            Integer deleted = (int) call.getLong(3);
+            if (LOG.isLoggable(Level.FINEST)) {
+                LOG.finest("Deleted from [" + eraseDto.getResourceType() + "/" + eraseDto.getLogicalId() + "] deleted [" + deleted + "]");
+            }
+
+            eraseRecord.setTotal(deleted);
+
+            ResourceEraseRecord.Status status = ResourceEraseRecord.Status.DONE;
+            if (deleted == -1) {
+                status = ResourceEraseRecord.Status.NOT_FOUND;
+            }
+            eraseRecord.setStatus(status);
+        } catch (SQLException x) {
+            LOG.log(Level.SEVERE, "SQL Call to Erase Failed to execute", x);
+            throw translator.translate(x);
+        }
+    }
+
+    /**
+     * Executes the SQL logic as part of the dao rather than via a stored procedure/function.
+     *
+     * @throws SQLException
+     */
+    public void runInDao() throws SQLException {
+        String resourceType = eraseDto.getResourceType();
+        String logicalId = eraseDto.getLogicalId();
+
+        long resourceTypeId = -1;
+        long logicalResourceId = -1;
+        long resourceId = -1;
+        int version = -1;
+        Integer total = 0;
+
+        // Prep 1: Get the v_resource_type_id
+        final String GET_RESOURCE_TYPE_ID =
+                "SELECT resource_type_id" +
+                        "  FROM resource_types" +
+                        "  WHERE resource_type = ?";
+        try (PreparedStatement stmt = getConnection().prepareStatement(GET_RESOURCE_TYPE_ID)) {
+            stmt.setString(1, resourceType);
+            ResultSet rs = stmt.executeQuery();
+            if (rs.next()) {
+                resourceTypeId = rs.getLong(1);
+            }
+        } catch (SQLException x) {
+            LOG.log(Level.SEVERE, GET_RESOURCE_TYPE_ID, x);
+            throw translator.translate(x);
+        }
+
+        // Prep 2: Get the logical from the system-wide logical resource level
+        final String GET_LOGICAL_RESOURCES_SYSTEM =
+                "SELECT logical_resource_id " +
+                        "  FROM logical_resources" +
+                        "  WHERE resource_type_id = ? AND logical_id = ?";
+        // Note the addForUpdate is important on the logical_resources query
+        try (PreparedStatement stmt = getConnection().prepareStatement(translator.addForUpdate(GET_LOGICAL_RESOURCES_SYSTEM))) {
+            stmt.setLong(1, resourceTypeId);
+            stmt.setString(2, logicalId);
+            ResultSet rs = stmt.executeQuery();
+            if (rs.next()) {
+                logicalResourceId = rs.getLong(1);
+            }
+        } catch (SQLException x) {
+            LOG.log(Level.SEVERE, GET_RESOURCE_TYPE_ID, x);
+            throw translator.translate(x);
+        }
+
+        /// Not found, return and stop processing.
```

Comment:
super minor
```suggestion
        // Not found, return and stop processing.
```

Reply:
Catching this locally and pushing in next commit.